#!/usr/bin/env python
# -*- coding: utf-8 -*-
PACKAGE = "zaryabot_bringup"

from dynamic_reconfigure.parameter_generator_catkin import*


gen = ParameterGenerator()

gen.add("kp_offset_line",   double_t , 0 ,"the proportion controller of  offset direction" , 0.5 , 0 , 5.0)
gen.add("kd_offset_line",   double_t , 0 ,"the differential controller of offset direction" , 0.0 ,0 , 5.0)
gen.add("kp_angular_line",  int_t , 0 ,"the proportion controller of z" , 30 , 0 , 500)
gen.add("kd_angular_line",  int_t , 0 ,"the proportion controller of z" , 0 , 0 , 30)
gen.add("velocity_forward_line", int_t , 0 ,"the velocity of going ahead in line-follwing", 70, 0 , 200)
gen.add("kp_forward_redpoint",  double_t , 0 ,"the proportion controller of going forward in redpoint " , 0.5 , 0 , 5.0)
gen.add("kd_forward_redpoint", double_t , 0 ,"the differential controller of going forward in red_point ", 0.0 , 0 , 5.0)
gen.add("kp_offset_redpoint",  double_t , 0 , "the proportion controller of offset in redpoint " ,0.5 , 0 ,5.0)
gen.add("kd_offset_redpoint", double_t , 0 ,"the differantial controller of offset in redpoint", 0.0 , 0, 5.0)



exit(gen.generate(PACKAGE, "dynamic_parameter","parameter"))
